docs/xml-dictionary.md

```diff
@@ -5,28 +5,352 @@ The dictionaries are stored in an *ad hoc* dictionary format. There
 appears to be many different variants of the format originally designed
 in Giellatekno for Northern Saami.
 
-
 The original documentation for this format is [here][original-docs], but
 it's in Norwegian. There's also a tiny bit of documentation inline in
 [the DTD file][gt_dictionary.dtd], but, as it claims:
 
 > This document is not quite finished yet...
 
-Here's an informal description of the XML format.
+This is a description of the XML format as it is used in itwêwina.
+A single dictionary XML file is a collection of **dictionary entries**
+(the _lexicon_) that have translations that are cited from one or more
+**dictionary sources**.
+
+The original XML dictionary format appears to support more use cases,
+but we have opted to focus on a few key use cases pertinent to Plains
+Cree.
+
+
+## `<r>`: The root element
+
+The root element is `<r>`. `<r>` contains:
+
+ - zero or more `<source>` elements
+ - zero or more `<e>` elements
+
+By convention, all `<source>` elements go near the top of the file, then
+the `<e>` elements follow alphabetically (according to their lemma).
+There will be far fewer `<source>` elements than `<e>` elements.
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<r>
+    <!-- dictionary sources -->
+    <source id="..."> ... </source>
+    <source id="..."> ... </source>
+    <source id="..."> ... </source>
+
+    <!-- dictionary entries -->
+    <e> ... </e>
+    <e> ... </e>
+    <e> ... </e>
+</r>
+```
+
+
+## `<source>`: Dictionary source
+
+A **dictionary source** is the metadata for a particular source of
+definitions. Think of it like a bibliography entry.
+
+`<source>` contains one required attribute, `id=""`, and exactly one
+required child: `<title>`.
+
+### Required attributes
+
+ - `id=""` a machine-readable ID used by `<t>` elements to "cite" which
+   source it comes from. This is similar to `id=""` attributes in HTML.
+   The ID **SHOULD** be a short, ASCII-representable string, containing
+   only alphanumerics (U+0030—U+0039, U+0041–U+005A, U+0061–U+007A) and
+   the hyphen-minus character (U+002D). The ID **MUST NOT** contain any
+   spacing or format characters.
+
+### Required children
+
+ - `<title>`: contains the title of the dictionary source in plain text.
```

Comment:
We might want to allow for the URL to a website pertaining to the dictionary/source, e.g. with <url>, and the author(s) of the source with <author>.

Indeed, Wolvengrey's Dictionary has quite many subsources, which we might want to be prepared for. These may be individual texts, individual recordings, or academic articles (and there may be tens if not hundreds of these). So we might want to be prepared for more detailed source information than just CW, MD, and Elders.

Reply:
Yeah, I figure the `<source>` element could be like a mini-bibliography entry, with `<url>`, `<author>`, `<year>`, `<editor>`, etc. _and_ subsources!

We'll add to it as the need arises. For now, only `<title>` seems necessary.